src/Collections.ts

```diff
@@ -575,6 +581,40 @@ export abstract class ArrayQueryable<TElement>
             action(this.source[i], i);
         }
     }
+
+    public sequenceEqual(other: IQueryable<TElement>, comparer?: EqualityComparer<TElement>): boolean
+    {
+        if (other instanceof ArrayQueryable)
+        {
+            const thisArray = this.asArray();
+            const otherArray = other.asArray();
+
+            const thisCount = thisArray.length;
+            const otherCount = otherArray.length;
+
+            if (thisCount != otherCount)
+            {
+                return false;
+            }
+
+            if (!comparer)
+            {
+                comparer = StrictEqualityComparer<TElement>();
+            }
+
+            for (let i = 0;i<thisCount;i++)
```

Comment:
Those manual loops can be optimized by using built-in functions

```ts
return thisArray.every((e, i) => comparer(e, otherArray[i]));
```

Reply:
If it's performance you want than it would be a better idea to just leave it as a simple for loop. No unnecessary allocations and callback/function calls

Reply:
that's right, then:

```
for (let i = 0, count = thisArray.length; i < count; ++i)
```